scripts/new-package.ts

```diff
@@ -1,47 +1,124 @@
 /// <reference types="node" />
 
-import { mkdirSync, writeFileSync } from "fs";
-import * as path from "path";
+import { get } from "http"
+import { mkdir, writeFile, unlink, rmdir } from "fs"
+import * as path from "path"
 
-const newPackageName = process.argv[2];
+interface Package {
+    name: string
+    description: string
+    version: string
+    homepage?: string
+}
+
+interface Registry {
+    name: string
+    description: string
+    "dist-tags": { latest: string }
+    versions: { [version: string]: Package }
+}
+
+const newPackageName = process.argv[2]
 if (!newPackageName) {
     throw new Error("Usage: node scripts/new-package.js new-package-name")
 }
 
-mkdirSync(newPackageName);
+run().catch(e => {
+    console.error(e)
+    process.exit(1)
+})
+
+async function run() {
+    const files: [string, string][] = [
+        ["index.d.ts", await getIndex()],
+        [`${newPackageName}-tests.ts`, ""],
+        ["tsconfig.json", `${JSON.stringify(getTSConfig(), undefined, 4)}\n`],
+        ["tslint.json", '{ "extends": "../tslint.json" }\n'],
+    ]
+
+    try {
+        await pify1(mkdir, newPackageName)
+        await Promise.all(files.map(([name, text]) => write(name, text)))
+    } catch (e) {
+        if (e.code === "EEXIST") {
+            console.warn(`Module “${newPackageName}” already exists!`)
+        } else {
+            console.error(`Error creating module files: ${e}\nCleaning Up.`)
+            await Promise.all(files.map(([name]) => rm(name)))
+            await pify1(rmdir, newPackageName)
+        }
+        process.exit(1)
+    }
+}
+
+async function getIndex() {
+    let version = "x.x"
+    let project = "https://github.com/baz/foo (Does not have to be to GitHub, but prefer linking to a source code repository rather than to a project website.)"
+    try {
+        const pkg = await new Promise<Registry>((resolve, reject) => {
+            get(`http://registry.npmjs.org/${newPackageName}`, (res) => {
+                if (res.statusCode !== 200) return reject(res.statusCode)
```

Comment:
Should wrap this in an error; the `catch` extracts `stack`, which is undefined for a number.